spikes/lk_led_animations/main.cpp

```diff
@@ -0,0 +1,116 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "PinNames.h"
+
+#include "drivers/HighResClock.h"
+#include "rtos/ThisThread.h"
+
+#include "CoreLED.h"
+#include "CoreSPI.h"
+#include "HelloWorld.h"
+#include "LedKit.h"
+#include "LogKit.h"
+#include "animations/BlinkGreen.h"
+#include "animations/Fire.h"
+#include "animations/Rainbow.h"
+#include "animations/SpinBlink.h"
+#include "animations/Sprinkles.h"
+
+using namespace leka;
+using namespace std::chrono;
+
+//
+// MARK: - Global definitions
+//
+
+namespace {
+
+namespace leds {
+
+	namespace internal {
+
+		namespace ears {
+
+			auto spi			= CoreSPI {LED_EARS_SPI_MOSI, NC, LED_EARS_SPI_SCK};
+			constexpr auto size = 2;
+
+		}	// namespace ears
+
+		namespace belt {
+
+			auto spi			= CoreSPI {LED_BELT_SPI_MOSI, NC, LED_BELT_SPI_SCK};
+			constexpr auto size = 20;
+
+		}	// namespace belt
+
+		namespace animations {
+
+			auto thread		 = rtos::Thread {};
+			auto event_queue = events::EventQueue {};
+
+		}	// namespace animations
+
+	}	// namespace internal
+
+	auto ears = CoreLED<internal::ears::size> {internal::ears::spi};
+	auto belt = CoreLED<internal::belt::size> {internal::belt::spi};
+
+	auto kit = LedKit {internal::animations::thread, internal::animations::event_queue, ears, belt};
```

Comment:
This refactorization is different from `spike/lk_led_kit`, definition of namespace and their orders risk to be inconstant among different `main.cpp`.

```cpp
namespace leds {

namespace spi {

	auto belt = CoreSPI {LED_BELT_SPI_MOSI, NC, LED_BELT_SPI_SCK};
	auto ears = CoreSPI {LED_EARS_SPI_MOSI, NC, LED_EARS_SPI_SCK};

}	// namespace spi

namespace animations {

	auto thread		 = rtos::Thread {};
	auto event_queue = events::EventQueue {};

}	// namespace animations

auto ears = CoreLED<LedKit::kNumberOfLedsEars> {spi::ears};
auto belt = CoreLED<LedKit::kNumberOfLedsBelt> {spi::belt};

}	// namespace leds
```